### ISSUE TITLE
Automated Domain Extraction from URL

### ISSUE THREAD

**Issue by megan201296** [CERT-BDF/TheHive#46]
# Automated Domain Extraction from URL

### Request Type  
Feature Request  
  
### Request Details     
When a URL is added to the observables tab it should also automatically extract the domain from the URL and add it as a domain observable. Preferably it should also either not add or throw an error if that domain is already added to the case. This assists in automation and increases chances of missing correlating indicators between cases.

**Comment by megan201296** [comment]
Just a related thought, though may require a new feature request/enhancement case... When a file is uploaded, add the hashes that were generated as separate observables as well.

**Comment by saadkadhi** [comment]
Hi @megan201296,

Thank you for your feature request. 

We implemented this feature in early versions of TheHive but we removed it as there was a serious debate among us since we were adding new observables that we didn't... well... observe. To an extent, it's like making inferences when performing forensics. 

But we hear you and we understand that there might be a need for domain extraction out of observables which datatype is `url`. Hence we will be moving this feature to a domain extractor analyzer. We will also implement a setting that will let you run automatically analyzers on observables according to their datatype.

Hence, if you turn on the **run automatically** setting for the domain extractor analyzer, whenever you'll add a URL, the domain will be extracted and added to the case.

Would that be helpful?

PS: please create a separate feature request for the addition of the hashes computed from a `file` observable to the case. We will treat it in the same fashion: through an analyzer. I hope you noted that file hashes are already computed.

**Comment by megan201296** [comment]
Yes. That sounds perfect! And I'll submit the file hash extractor separately. I did notice that they are automatically computed and added to the file observable details. However, right now I manually pull the hashes from there and add them as new observables also. Would be nice for some automation there since it is already computed; just push it to the observable tab.

**Comment by megan201296** [comment]
@saadkadhi don't forget to close this one with the the release of Cerana 1 since #209 covers this :)
